pkg/cmd/repo/rename/rename.go

```
@@ -0,0 +1,125 @@
+	}
+
+	cmd := &cobra.Command{
+		Use:   "rename [<new-name>]",
+		Short: "Rename a repository",
+		Long: heredoc.Doc(`Rename a GitHub repository
+
+		By default, renames the current repository otherwise rename the specified repository.`),
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.BaseRepo = f.BaseRepo
+			opts.HasRepoOverride = cmd.Flags().Changed("repo")
+
+			if len(args) > 0 {
+				opts.newRepoSelector = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return cmdutil.FlagErrorf("could not prompt: new name required when not running interactively")
+			}
+
+			if runf != nil {
+				return runf(opts)
+			}
+			return renameRun(opts)
+		},
+	}
+
+	cmdutil.EnableRepoOverride(cmd, f)
+
+	return cmd
+}
+
+func renameRun(opts *RenameOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	var newRepo ghrepo.Interface
+	var baseRemote *context.Remote
+	newRepoName := opts.newRepoSelector
+
+	currRepo, err := opts.BaseRepo()
+	if err != nil {
+		return err
+	}
+
+	if newRepoName == "" {
+		err = prompt.SurveyAskOne(
+			&survey.Input{
+				Message: fmt.Sprintf("Rename %s to: ", ghrepo.FullName(currRepo)),
+			},
+			&newRepoName,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	newRepo = ghrepo.NewWithHost(currRepo.RepoOwner(), newRepoName, currRepo.RepoHost())
+
+	err = runRename(httpClient, currRepo, newRepoName)
+	if err != nil {
+		return fmt.Errorf("API called failed: %s", err)
+	}
+
+	if opts.IO.IsStdoutTTY() {
+		cs := opts.IO.ColorScheme()
+		fmt.Fprintf(opts.IO.Out, "%s Renamed repository %s\n", cs.SuccessIcon(), ghrepo.FullName(newRepo))
+	}
+
+	if !opts.HasRepoOverride {
+		cs := opts.IO.ColorScheme()
+		cfg, err := opts.Config()
+		if err != nil {
+			return err
+		}
+
+		protocol, _ := cfg.Get(currRepo.RepoHost(), "git_protocol")
+		remotes, _ := opts.Remotes()
+		baseRemote, _ = remotes.FindByRepo(currRepo.RepoOwner(), currRepo.RepoName())
```

Each of these functions can return an error, if any of them do and we try to continue without checking them we risk the command blowing up. Better to check if the error is not nil before continuing.

```golang
if err != nil {
     return err
}
```

---

I tagged you in a earlier comment where we decided to ignore the errors for updating the repo

---

We did specify before that maybe the errors should not bubble up to the user and instead a generic warning is given saying the remote was unable to be updated, which is still valid but if we don't check for errors here the program will actually crash not fail gracefully.

Imagine if `baseRemote` is nil here because there was an error, when we do `baseRemote.Name` on line 116 there will be a `nil` exception and a stack trace will be output to the user.